zio-http/src/main/scala/zio/http/netty/client/ClientInboundHandler.scala

```diff
@@ -55,7 +55,7 @@ final class ClientInboundHandler(
         ctx.writeAndFlush(fullRequest)
       case _: HttpRequest               =>
         ctx.write(jReq)
-        NettyBodyWriter.writeAndFlush(req.body, ctx).foreach { effect =>
+        NettyBodyWriter.writeAndFlush(req.body, None, ctx).foreach { effect =>
```

Comment:
slightly tangential to the purpose of this PR, but it would be nice if the client side mirrored the server side with respect to content-length. specifically, if the request encoding path did not adjust the content-length and transfer-encoding headers so that the content-length header could be checked here and streaming bodies sent the same way for both client and server